gensim/models/fasttext.py

```diff
@@ -744,54 +766,52 @@ def init_ngrams_weights(self, wv, update=False, vocabulary=None):
 
         """
         if not update:
-            wv.ngrams = {}
             wv.vectors_vocab = empty((len(wv.vocab), wv.vector_size), dtype=REAL)
             self.vectors_vocab_lockf = ones((len(wv.vocab), wv.vector_size), dtype=REAL)
 
             wv.vectors_ngrams = empty((self.bucket, wv.vector_size), dtype=REAL)
             self.vectors_ngrams_lockf = ones((self.bucket, wv.vector_size), dtype=REAL)
 
-            all_ngrams = []
-            for w, ngrams in iteritems(wv.ngrams_word):
-                all_ngrams += ngrams
-
-            all_ngrams = list(set(all_ngrams))
-            wv.num_ngram_vectors = len(all_ngrams)
-            logger.info("Total number of ngrams is %d", len(all_ngrams))
-
             wv.hash2index = {}
+            wv.buckets_word = {}
             ngram_indices = []
             new_hash_count = 0
-            for i, ngram in enumerate(all_ngrams):
-                ngram_hash = _ft_hash(ngram) % self.bucket
-                if ngram_hash in wv.hash2index:
-                    wv.ngrams[ngram] = wv.hash2index[ngram_hash]
-                else:
-                    ngram_indices.append(ngram_hash % self.bucket)
-                    wv.hash2index[ngram_hash] = new_hash_count
-                    wv.ngrams[ngram] = wv.hash2index[ngram_hash]
-                    new_hash_count = new_hash_count + 1
+            wv.num_ngram_vectors = 0
```

Comment:
We could probably reduce some variables here - there seems to be some redundancy, if I understand correctly. `wv.num_ngram_vectors`, `new_hash_count` and `len(ngram_indices)` serve effectively the same purpose.
Maybe we could use `len(ngram_indices)` within the loop and set `wv.num_ngram_vectors` at the end of the loop?